src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = findIndex(hash(key));
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> iterator = table[index];
+            while (iterator.next != null || Objects.equals(iterator.key, key)) {
+                if (Objects.equals(iterator.key, key)) {
+                    iterator.value = newNode.value;
+                    return;
+                }
+                iterator = iterator.next;
+            }
+            iterator.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = findIndex(hash(key));
+        Node<K, V> iterator = table[index];
+        while (iterator != null) {
+            if (Objects.equals(iterator.key, key)) {
+                return iterator.value;
+            }
+            iterator = iterator.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkSize() {
+        if (size == threshold) {
+            resize();
+        }
+    }
```

Comment:
do we need to move this to a separate private method? the amount of code is so small

Reply:
You are right